app/assets/javascripts/editor.js

```diff
@@ -23,131 +26,122 @@ $E = {
-    var replace = a + sel + b;
-    if (newlineParameterExists) {
-      replace = replace + "\n\n";
-    }
-    if (newlineParameterExists && $E.textarea[0].selectionStart > 0) { 
-      replace = "\n" + replace; 
-    }
-    $E.textarea.val($E.textarea.val().substring(0,start) + replace + $E.textarea.val().substring(end,len));
-  },
-  bold: function() {
+  wrap(a, b, newlineDesired = false, fallback) {
+    // we only refresh $E's values if we are on a page with multiple comments
+    if (!this.isSingleFormPage(window.location.pathname)) { this.refresh(); }
+
+    const selectionStart = $E.textarea[0].selectionStart;
+    const selectionEnd = $E.textarea[0].selectionEnd;
+    const selection = fallback || $E.textarea.val().substring(selectionStart, selectionEnd); // fallback if nothing has been selected, and we're simply dealing with an insertion point
+
+    let newText = a + selection + b; // ie. ** + selection + ** (wrapping selection in bold)
+    if (newlineDesired) { newText = newText + "\n\n"; }
+    const selectionStartsMidText = $E.textarea[0].selectionStart > 0;
+    if (newlineDesired && selectionStartsMidText) { newText = "\n" + newText; }
+
+    const textLength = $E.textarea.val().length;
+    const textBeforeSelection = $E.textarea.val().substring(0, selectionStart);
+    const textAfterSelection = $E.textarea.val().substring(selectionEnd, textLength);
+    $E.textarea.val(textBeforeSelection + newText + textAfterSelection);
+  }
+  bold() {
     $E.wrap('**','**')
-  },
-  italic: function() {
+  }
+  italic() {
     $E.wrap('_','_')
-  },
-  link: function(uri) {
+  }
+  link(uri) {
     uri = prompt('Enter a URL');
     if (uri === null) { uri = ""; }
     $E.wrap('[', '](' + uri + ')');
-  },
-  image: function(src) {
+  }
+  image(src) {
     $E.wrap('\n![',']('+src+')\n')
-  },
-  h1: function() {
-    $E.wrap('#','')
-  },
-  h2: function() {
+  }
+  // these header formatting functions are not used anywhere, so commenting them out for now to pass codeclimate:
+
+  // h1() {
+  //   $E.wrap('#','')
+  // }
+  h2() {
     $E.wrap('##','')
-  },
-  h3: function() {
-    $E.wrap('###','')
-  },
-  h4: function() {
-    $E.wrap('####','')
-  },
-  h5: function() {
-    $E.wrap('#####','')
-  },
-  h6: function() {
-    $E.wrap('######','')
-  },
-  h7: function() {
-    $E.wrap('#######','')
-  },
+  }
+  // h3() {
+  //   $E.wrap('###','')
+  // }
+  // h4() {
+  //   $E.wrap('####','')
+  // }
+  // h5() {
+  //   $E.wrap('#####','')
+  // }
+  // h6() {
+  //   $E.wrap('######','')
+  // }
+  // h7() {
+  //   $E.wrap('#######','')
+  // }
   // this function is dedicated to Don Blair https://github.com/donblair
-  save: function() {
-    localStorage.setItem('plots:lastpost',$E.textarea.val())
-    localStorage.setItem('plots:lasttitle',$E.title.val())
```

Comment:
Deleted the `lasttitle`, it seems to be unused. Will be refactoring this code anyway in the next few weeks.